src/DynamoCoreWpf/ViewModels/PackageManager/PackagePathViewModel.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using Dynamo.Interfaces;
+using DelegateCommand = Dynamo.UI.Commands.DelegateCommand;
+
+
+namespace Dynamo.ViewModels
+{
+    class PackagePathViewModel : ViewModelBase
+    {
+        public ObservableCollection<string> RootLocations { get; private set; }
+        private int selectedIndex;
+        public int SelectedIndex {
+            get { return selectedIndex; }
+            set 
+            {
+                selectedIndex = value;
+                if (selectedIndex < RootLocations.Count - 1)
+                    MovePathDownCommand.RaiseCanExecuteChanged();
+                if (selectedIndex > 0)
+                    MovePathUpCommand.RaiseCanExecuteChanged();
+            }
+        }
+
+        private readonly IPreferences setting;
+
+
+        public DelegateCommand AddPathCommand { get; private set; }
+        public DelegateCommand DeletePathCommand { get; private set; }
+        public DelegateCommand MovePathUpCommand { get; private set; }
+        public DelegateCommand MovePathDownCommand { get; private set; }
+        public DelegateCommand UpdatePathCommand { get; private set; }
+        public DelegateCommand SaveSettingCommand { get; private set; }
+
+        public PackagePathViewModel(IPreferences setting)
+        {
+            RootLocations = new ObservableCollection<string>(setting.CustomPackageFolders);
+            SelectedIndex = 0;
+            this.setting = setting;
+
+            AddPathCommand = new DelegateCommand(p => InsertPathAt(p as string, RootLocations.Count));
+            DeletePathCommand = new DelegateCommand(p => RemovePathAt(SelectedIndex), CanDelete);
+            MovePathUpCommand = new DelegateCommand(p => SwapPath(SelectedIndex, SelectedIndex - 1), CanMoveUp);
+            MovePathDownCommand = new DelegateCommand(p => SwapPath(SelectedIndex, SelectedIndex + 1), CanMoveDown);
+            UpdatePathCommand = new DelegateCommand(p => UpdatePathAt(p as string, SelectedIndex));
```

Comment:
The above five commands must be updated to have this signature:

```csharp
new DelegateCommand(p => SwapPath(((int) p), ((int) p) + 1), CanMoveDown);
```

Which means each `DelegateCommand` has an input as `object` but is actually an `int` (the selected item's index).